module/zfs/spa.c

```diff
@@ -4945,7 +4945,9 @@ spa_vdev_detach(spa_t *spa, uint64_t guid, uint64_t pguid, int replace_done)
 	 * But first make sure we're not on any *other* txg's DTL list, to
 	 * prevent vd from being accessed after it's freed.
 	 */
-	vdpath = spa_strdup(vd->vdev_path);
```

Comment:
@dinatale2 @heary-cao I think it would be simpler simple make the following one line change to this function to handle this case.

```
-	vdpath = spa_strdup(vd->vdev_path);
+	vdpath = spa_strdup(vd->vdev_path ? vd->vdev_path : "none");
```

This ensures that when that `vdpath` will always be non-NULL regardless of what `vd->vdev_path` contains and ensure we always generate a useful message for the history.